libs/community/langchain_community/chat_models/writer.py

```diff
@@ -149,45 +92,122 @@ def _identifying_params(self) -> Dict[str, Any]:
         }
 
         return ChatResult(generations=generations, llm_output=llm_output)
 
-    def _convert_messages_to_dicts(
+    @staticmethod
+    def _convert_langchain_to_writer(message: BaseMessage) -> dict:
+        """Convert a LangChain message to a Writer message dict."""
+        message_dict = {"role": "", "content": message.content}
+
+        if isinstance(message, ChatMessage):
+            message_dict["role"] = message.role
+        elif isinstance(message, HumanMessage):
+            message_dict["role"] = "user"
+        elif isinstance(message, AIMessage):
+            message_dict["role"] = "assistant"
+            if message.tool_calls:
+                message_dict["tool_calls"] = [
+                    {
+                        "id": tool["id"],
+                        "type": "function",
+                        "function": {"name": tool["name"], "arguments": tool["args"]},
+                    }
+                    for tool in message.tool_calls
+                ]
+        elif isinstance(message, SystemMessage):
+            message_dict["role"] = "system"
+        elif isinstance(message, ToolMessage):
+            message_dict["role"] = "tool"
+            message_dict["tool_call_id"] = message.tool_call_id
+        else:
+            raise ValueError(f"Got unknown message type: {type(message)}")
+
+        if message.name:
+            message_dict["name"] = message.name
+
+        return message_dict
+
+    @staticmethod
+    def _convert_writer_to_langchain(response_message: Any) -> BaseMessage:
+        """Convert a Writer message to a LangChain message."""
+        if not isinstance(response_message, dict):
+            response_message = json.loads(
+                json.dumps(response_message, default=lambda o: o.__dict__)
+            )
+
+        role = response_message.get("role", "")
+        content = response_message.get("content")
+        if not content:
+            content = ""
+
+        if role == "user":
+            return HumanMessage(content=content)
+        elif role == "assistant":
+            additional_kwargs = {}
+            if tool_calls := response_message.get("tool_calls", []):
+                additional_kwargs["tool_calls"] = tool_calls
+            return AIMessageChunk(content=content, additional_kwargs=additional_kwargs)
+        elif role == "system":
+            return SystemMessage(content=content)
+        elif role == "tool":
+            return ToolMessage(
+                content=content,
+                tool_call_id=response_message.get("tool_call_id", ""),
+                name=response_message.get("name", ""),
+            )
+        else:
+            return ChatMessage(content=content, role=role)
+
+    def _convert_messages_to_writer(
         self, messages: List[BaseMessage], stop: Optional[List[str]] = None
     ) -> Tuple[List[Dict[str, Any]], Dict[str, Any]]:
+        """Convert a list f LandChain messages to List of Writer dict messages."""
```

Comment:
?

Reply:
Ohh, silly typing mistake, my bad, I'm sorry! I'll fix this